conjure-lib/src/test/java/com/palantir/conjure/java/lib/internal/ConjureCollectionsTest.java

```diff
@@ -82,4 +86,20 @@ public void newNonNullSafeLongList() {
         safeLongList.clear();
         assertThat(safeLongList).hasSize(0);
     }
+
+    @Test
+    public void testSerializationRoundtrip() throws JsonProcessingException {
+        List<Double> expectedDoubles = List.of(1.1, 2.2, 3.3);
+        testSerialization(ConjureCollections.newNonNullDoubleList(expectedDoubles), expectedDoubles);
+        List<Integer> expectedIntegers = List.of(1, 2, 3);
+        testSerialization(ConjureCollections.newNonNullIntegerList(expectedIntegers), expectedIntegers);
+        List<SafeLong> expectedSafeLongs = List.of(SafeLong.of(1L), SafeLong.of(2L), SafeLong.of(3L));
+        testSerialization(ConjureCollections.newNonNullSafeLongList(expectedSafeLongs), expectedSafeLongs);
+    }
```

Comment:
serialization of the `ConjureCollections.newNonNull*List` isn't representative of usage within generated objects. Generated objects wrap these with `Collections.unmodifiableList`, so the type of the list that Jackson perceives is `java.util.Collections$UnmodifiableList` or `java.util.Collections$UnmodifiableRandomAccessList`, and the custom serializer isn't called.